tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -10,14 +10,15 @@ load "${BATS_TEST_DIRNAME}/confidential_common.sh"
 
 setup() {
     confidential_setup || skip "Test not supported for ${KATA_HYPERVISOR}."
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
```

Comment:
We need this because the normal aks tests are still calling this and might run with a supported non-tee HV but without the snapshotter, right?

Reply:
Yes, I would say, instead of Wainer, with an example for a workflow `run-k8s-tests-on-zvsi` which uses `devmapper`:

```
1..4
ok 1 Test we can pull an unencrypted image outside the guest with runc and then inside the guest successfully # skip None snapshotter was found but this test requires one
# [bats-exec-test:61] INFO: Need to apply image annotations
ok 2 Test we can pull a large image inside the guest # skip None snapshotter was found but this test requires one
# [bats-exec-test:61] INFO: Need to apply image annotations
ok 3 Test we can pull an unencrypted image inside the guest twice in a row and then outside the guest successfully # skip None snapshotter was found but this test requires one
# [bats-exec-test:61] INFO: Need to apply image annotations
ok 4 Test we can pull an other unencrypted image outside the guest and then inside the guest successfully # skip None snapshotter was found but this test requires one
```